pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ConnectionsLimiter.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.service;
+
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.regex.Pattern;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.mutable.MutableInt;
```

Comment:
Why not just use an Integer or something standard in the java library?

Reply:
IMO,
1  API is more convenient
2 Modifying the value of an object does not require recreating a new object

Reply:
Doesn't the code create a new object everytime anyway for the `CONNECTIONS.putIfAbsent(ip, new MutableInt(0));`?  IMO adding an external library where standard objects and primitives works seems unnecessary (it's an unneeded dependency). And looking at the code this api the `MutableInt` makes the code more difficult for me to follow.

```

```
is much clearer than
```
final int connectionsPerIP = CONNECTIONS.getOrDefault(ip, 0);
// make checks
CONNECTIONS.put(ip, connectionsPerIP++)
```